server/exercise/src/main/R/classification/tap.R

```diff
@@ -0,0 +1,52 @@
+library(emuR)
+library(FactoMineR)
+
+# Function to extract a collection of moving windows (of size `size`) from a example (time series) data.
+#
+# @param data time series data over which we will move our sampling window
+# @param size size of sampling window
+# @param inc  increment by which we move sampling window (default is 10 events)
+# @returns    list of sampling windows extracted from the time series data (sample window includes data start and stop
+#             indexes)
+windowSampling = function(data, size, inc = 10) {
+  dataSize = dim(data)[1]
+  startIndexes = seq(1, dataSize, by=inc)
+  lapply(startIndexes, function(index) { list(index, index+size-1, data[index:(index+size-1),]) })
+}
+
+# Function to split an input CSV file (holding accelerometer data) into a (saved) collection of sampling windows.
+#
+# @param input
+# @param size
+# @param inc
+splitInput = function(input, size, inc = 10) {
+  input.csv = read.csv(file=input, col.names=c("x", "y", "z"))
+  baseFilename = sub("\\.csv", "", input)
+  for (window in windowSampling(input.csv, size, inc)) {
+    write.table(window[3], file=paste(baseFilename, "-", window[1], "-", window[2], ".csv", sep=""), sep=",", row.names=FALSE, col.names=FALSE)
```

Comment:
I think we are looking for a format of data similar to 
`label, feature1, feature2, feature3, ...`
that could be directly fed as an input to a classifier.

E.g. transforming files in format
`x1,y1,z1`
`x2,y2,z2`
to 
`l1,x1,y1,z1,x2,y2,z2, ...`
or 
`l1,x1,x2,...y1,y2,...z1,z2...`